hashintel/hash
# H-198: Add Linear connection webhook handler

## 🌟 What is the purpose of this PR?

Add a handler for Linear webhook events. This just receives the events at the moment and doesn't do anything with them.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:



### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues

- Testing this locally relies on having a webhook URL registered with the Linear app that can reach the local machine, e.g. via an ngrok tunnel. 


## 🐾 Next steps

- Do something with the webhook events, e.g. create or update existing HASH entities which were ingested from Linear entities
- Automate the process of creating an ngrok tunnel to the local machine and registering that as a webhook listener with the Linear testing app



## ❓ How to test this?



1.  Create a Linear app in a testing org. Toggle on webhook first as doing this will clear the rest of the app settings form.
2. Create an ngrok tunnel to the API locally (`ngrok http 5001`) and set that URL as the webhook URL
3. set the webhook's secret as a `LINEAR_WEBHOOK_SECRET` env locally
4. Edit something in the Linear testing org (e.g. create a comment or edit an issue) and see the event logged
